### PR TITLE
Closes  #12162: Staff coords refactor

### LINKED ISSUE #12162: Use Coords for Staff functions
Some Staff functions still take individual `x` and `y` parameters. The goal is to make them use `CoordsXY`.

Note: You can do all of the functions in all PR (seems small overall), but please do each of them into a different commit.

- `staff_is_patrol_area_set` should receive a `const CoordsXY&` instead of `x`  and `y`.
- `staff_set_patrol_area` should receive a `const CoordsXY&` instead of `x`  and `y`.
- `staff_toggle_patrol_area` should receive a `const CoordsXY&` instead of `x`  and `y`.

Functions are at `src\openrct2\peep\Staff.cpp`

### PR DESCRIPTION
**Pull request by Jimver**
Hi all,

This PR implements the refactorings as listed in #12162

Do I need to bump the network version or is this not necessary?

### CODE REVIEW

File: src/openrct2/peep/Staff.cpp [hunk 1]
```diff
@@ -419,10 +419,10 @@ void staff_set_patrol_area(int32_t staffIndex, int32_t x, int32_t y, bool value)
     }
 }
 
-void staff_toggle_patrol_area(int32_t staffIndex, int32_t x, int32_t y)
+void staff_toggle_patrol_area(int32_t staffIndex, const CoordsXY& coords)
 {
-    x = (x & 0x1F80) >> 7;
-    y = (y & 0x1F80) >> 1;
+    int32_t x = (coords.x & 0x1F80) >> 7;
+    int32_t y = (coords.y & 0x1F80) >> 1;
```
Comment (tupaschoal) [inline src/openrct2/peep/Staff.cpp:425]:
```suggestion
    auto offsetCoords = CoordsXY{ (coords.x & 0x1F80) >> 7, (coords.y & 0x1F80) >> 1 };
```

File: src/openrct2/rct1/S4Importer.cpp [hunk 2]
```diff
@@ -1619,7 +1619,7 @@ class S4Importer final : public IParkImporter
                 x <<= 7;
                 int32_t y = val & 0x3E0;
                 y <<= 2;
-                staff_set_patrol_area(staffmember->StaffId, x, y, true);
+                staff_set_patrol_area(staffmember->StaffId, CoordsXY(x, y), true);
```
Comment (tupaschoal) [inline src/openrct2/rct1/S4Importer.cpp:1622]:
```suggestion
                staff_set_patrol_area(staffmember->StaffId, { x, y }, true);
```

File: src/openrct2/peep/Staff.cpp [hunk 3]
```diff
@@ -400,10 +400,10 @@ bool staff_is_patrol_area_set_for_type(STAFF_TYPE type, const CoordsXY& coords)
     return staff_is_patrol_area_set(STAFF_MAX_COUNT + type, coords.x, coords.y);
 }
 
-void staff_set_patrol_area(int32_t staffIndex, int32_t x, int32_t y, bool value)
+void staff_set_patrol_area(int32_t staffIndex, const CoordsXY& coords, bool value)
 {
-    x = (x & 0x1F80) >> 7;
-    y = (y & 0x1F80) >> 1;
+    int32_t x = (coords.x & 0x1F80) >> 7;
+    int32_t y = (coords.y & 0x1F80) >> 1;
```
Comment (tupaschoal) [inline src/openrct2/peep/Staff.cpp:406]:
```suggestion
    auto offsetCoords = CoordsXY{ (coords.x & 0x1F80) >> 7, (coords.y & 0x1F80) >> 1 };
```

File: src/openrct2/peep/Staff.cpp [hunk 4]
```diff
@@ -373,16 +373,16 @@ void staff_reset_stats()
...
     // Patrol quads are stored in a bit map (8 patrol quads per byte).
     // Each patrol quad is 4x4.
     // Therefore there are in total 64 x 64 patrol quads in the 256 x 256 map.
     // At the end of the array (after the slots for individual staff members),
     // there are slots that save the combined patrol area for every staff type.
 
-    x = (x & 0x1F80) >> 7;
-    y = (y & 0x1F80) >> 1;
+    int32_t x = (coords.x & 0x1F80) >> 7;
+    int32_t y = (coords.y & 0x1F80) >> 1;
```
Comment (tupaschoal) [inline src/openrct2/peep/Staff.cpp:385]:
```suggestion
    auto offsetCoords = CoordsXY{ (coords.x & 0x1F80) >> 7, (coords.y & 0x1F80) >> 1 };
```

**Comment by Jimver** [comment]
Thanks for the suggestions! I just now pushed the requested changes.

**Comment by duncanspumpkin** [comment]
I know you've just completed this but i was looking at it and i think we can do better. We have 3 functions that all do the same computation. This should be extracted out into a separate function.

```
std::pair<int32_t, int32_t> getPatrolAreaOffsetIndex(const CoordsXY& coords)
{
    auto hash = ((coords.x & 0x1F80) >> 7) |((coords.y & 0x1F80) >> 1);
    return { hash >> 5, hash & 0x1F };
}

auto [ offset, index ] = getPatrolAreaOffsetIndex(coords);
```

**Comment by tupaschoal** [comment]
I've extracted Duncan's comment into an issue, as tracked above.

**Pull request merged**